lib/features/docker_save.js

```diff
@@ -23,26 +27,81 @@ export default class DockerSave {
     let imgStream = await image.get();
     let zipStream = zlib.createGzip();
     imgStream.pipe(zipStream).pipe(fs.createWriteStream(pathname));
-    await waitForEvent(zipStream, 'end');
+    await new Promise((accept, reject) => {
+      zipStream.on('end', accept);
+      zipStream.on('error', (error) => reject(error));
+    });
     debug('tar written');
 
     let stat = await fs.stat(pathname);
     debug(stat.size);
     let uploadStream = fs.createReadStream(pathname);
-    let expiration = new Date(Date.now() + task.runtime.dockerSave.expiration);
+    let expiration = taskcluster.fromNow(task.runtime.dockerSave.expiration);
 
-    await uploadToS3(task, uploadStream, ARTIFACT_NAME, expiration, {
+    await uploadToS3(task, uploadStream, 'public/dockerImage.tar', expiration, {
       'content-type': 'application/x-tar',
       'content-length': stat.size,
       'content-encoding': 'gzip'
     });
 
-    debug('artifact uploaded');
+    debug('docker image uploaded');
 
     //cleanup
-    fs.unlink(pathname).catch(() => {
-      task.runtime.log('[alert-operator] could not delete docker save tarball, worker may run out of hdd space');
+    fs.unlink(pathname).catch((err) => {
+      task.runtime.log('[alert-operator] could not delete cache save tarball, worker may run out of hdd space\r\n'
+        + err + err.stack);
     });
     await image.remove();
   }
+
+  //uploading a cache folder as an artifact
+  async uploadCache(task, cacheStr) {
+    let cache = cacheStr.split(KEY_DELIMITER);
```

Comment:
I think in the long run, we might want to consider saving the bindings as an array of objects with 
`
{name: test-cache
slugid: ofdxyug043ntlky32
}
`
as entries.